client/python/gradio_client/utils.py

```diff
@@ -243,6 +244,7 @@ class Communicator:
     reset_url: str
     should_cancel: bool = False
     event_id: str | None = None
+    prediction_complete: bool = False
```

Comment:
This field is added so that we can detect when a prediction is complete in the `stream_sse_v1plus()` thread so that we can check this field from the `check_for_cancel()` thread and have it terminate itself (since threads can't just be canceled like asyncio tasks)

Reply:
Update: this field has been renamed to `thread_complete`